eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -375,8 +375,16 @@ object Router {
 
   case class FinalizeRoute(amount: MilliSatoshi, hops: Seq[PublicKey], assistedRoutes: Seq[Seq[ExtraHop]] = Nil)
 
-  case class RouteResponse(hops: Seq[ChannelHop], ignoreNodes: Set[PublicKey], ignoreChannels: Set[ChannelDesc], allowEmpty: Boolean = false) {
+  case class Route(amount: MilliSatoshi, hops: Seq[ChannelHop], allowEmpty: Boolean = false) {
```

Comment:
`allowEmpty` doesn't seem to be used?

Reply:
Unfortunately it is, it was added by the multipart payment FSM because the current split algorithm "forces" the first hop to the payment FSM. It's a bit of a hack, and will be removed in a few PRs for the split v2 ;)